YGOProAnalyticsServer/Services/Others/BanlistService.cs

```diff
@@ -16,15 +20,65 @@ namespace YGOProAnalyticsServer.Services.Others
     public class BanlistService : IBanlistService
     {
         readonly YgoProAnalyticsDatabase _db;
+        readonly IMemoryCache _cache;
+        readonly IAdminConfig _config;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="BanlistService"/> class.
         /// </summary>
         /// <param name="db">The database.</param>
-        /// <exception cref="ArgumentNullException">db</exception>
-        public BanlistService(YgoProAnalyticsDatabase db)
+        /// <param name="cache">The cache.</param>
+        /// <param name="config">The configuration.</param>
+        /// <exception cref="ArgumentNullException">
+        /// db
+        /// or
+        /// cache
+        /// or
+        /// config
+        /// </exception>
+        public BanlistService(
+            YgoProAnalyticsDatabase db,
+            IMemoryCache cache,
+            IAdminConfig config)
         {
             _db = db ?? throw new ArgumentNullException(nameof(db));
+            _cache = cache ?? throw new ArgumentNullException(nameof(cache));
+            _config = config ?? throw new ArgumentNullException(nameof(config));
+        }
+
+        /// <inheritdoc />
+        public async Task<IEnumerable<BanlistIdAndNameDTO>> GetListOfBanlistsNamesAndIdsAsNoTrackingFromCache(bool shouldIgnoreCache = false)
+        {
+            IEnumerable<BanlistIdAndNameDTO> dtos;
+            if (shouldIgnoreCache)
+            {
+                return await _getBanlistsIdAndNameDtosAsNoTracking();
+            }
+            else
+            if (!_cache.TryGetValue(CacheKeys.BanlistsIdAndNameDtos, out dtos))
+            {
+                dtos = await _getBanlistsIdAndNameDtosAsNoTracking();
+                var cacheOptions = new MemoryCacheEntryOptions()
+                    .SetPriority(CacheItemPriority.Normal)
+                    .SetSlidingExpiration(
+                        TimeSpan.FromHours(_config.BanlistSlidingCacheExpirationInHours))
+                    .SetAbsoluteExpiration(
+                        TimeSpan.FromHours(_config.BanlistAbsoluteCacheExpirationInHours));
+
+                _cache.Set(CacheKeys.BanlistsIdAndNameDtos, dtos, cacheOptions);
+            }
+
+            return dtos;
+        }
+
+        private async Task<IEnumerable<BanlistIdAndNameDTO>> _getBanlistsIdAndNameDtosAsNoTracking()
+        {
+             var z = await _db
```

Comment:
make the variable name mean something... or return the result instead of that variable.